Makefile

```diff
@@ -21,11 +21,15 @@ SOROBAN_RPC_GIT_REF=https://github.com/stellar/soroban-tools.git\#main
 SOROBAN_CLI_GIT_REF=https://github.com/stellar/soroban-tools.git\#main
 GO_GIT_REF=https://github.com/stellar/go.git\#soroban-xdr-next
 QUICKSTART_GIT_REF=https://github.com/stellar/quickstart.git\#master
-JS_SOROBAN_CLIENT_NPM_VERSION=^0.4.0
+JS_SOROBAN_CLIENT_NPM_VERSION=https://github.com/stellar/js-soroban-client.git\#main
 
-NON_AMD_ARCH=false 
-ifneq ($(shell uname -p),x86_64)
-	NON_AMD_ARCH=true
+# Override the docker image platform to build
+PLATFORM=
+
+ifdef PLATFORM
```

Comment:
we're you able to get this to work as cross-platform build locally?

this multi-arch has been travelled a bit, in past attempts, have observed this will run into an issue when specifying a docker platform that is different from the host platform, docker's qemu layer kicks in and triggers issues, specifically on an linux/arm64 host trying to build with  docker platform override of linux/amd64, the cargo commands grind to a slow halt and in some cases logging out numerous warning incompatibilities during compilations and never finish.

just curious, why do you need to cross compile the image, as the use case of this dockerfile is intended to be built and run on same host, defaulting to host platform at all times.

Reply:
the `NON_AMD_ARCH` reference should be deleted, I missed that from prior attempt to do something similar on cross builds, as mentioned, ran into issues on qemu.

Reply:
I built a `linux/amd64` image on my arm mac, so seems like it worked. I didn't test it extensively though. Happy to leave this out if you'd prefer.

Reply:
ok, maybe we can discuss more, would like to know how long it took, maybe it was specific to docker for mac version i used. 

```
Server: Docker Desktop 4.16.2 (95914)
 Engine:
  Version:          20.10.22
```

assume you have buildkit enabled by default on your docker engine?

Reply:
were you able to run that amd64 image on your arm mac, with --platform linux/amd64 also, that runs additional cargo compilations also, curious if those worked.